### DISCUSSION TITLE
Detect plagiarism with "known solutions", but not between known solutions

### DISCUSSION THREAD

**Discussion by a3nm** [jplag/JPlag#431] [Ideas]
Hi,

This issue concerns a feature which I think does not exist (so it would be a feature request), but maybe it is already possible to do this and I missed it. If so, sorry about that!

I use jPlag to find overlap in source files submitted by students in response to an assignment. For the assignment in question, I also have "reference code" that the students may have plagiarized: these include source files submitted by students from previous years, and assignment solutions that have been posted online.

I would like to detect overlaps between pairs of submitted source files, and between any submitted source file and reference source file. I am not interested in overlap between pairs of reference source files (which includes past plagiarism cases, past false alarms, etc.).

Currently, I do this by running jPlag over all code, with a special naming to distinguish between the submitted source files and reference source files. However, this is inconvenient, as the list of candidate matches is polluted by similarities between many pairs of reference source files.

My feature request would be to have a way to provide "reference code" as input, and have the submitted source files be analyzed for overlap with the reference source files (in addition to overlap between two submitted source files), but no analysis of overlap between two reference source files.

Thanks for your consideration and for your work in developing jPlag!

**Comment by tsaglam** [comment]
This is coming with the v4.0.0 release. It is already implemented with PR #289. You can test it if you build from the master branch and use the `-old` argument.
